include/rive/renderer.hpp

```diff
@@ -16,27 +17,66 @@ namespace rive
 {
 	class Vec2D;
 
+    using ColorInt = uint32_t;
+
 	enum class RenderPaintStyle
 	{
 		stroke,
 		fill
 	};
 
+    enum class RenderTileMode {
+        clamp,
+        repeat,
+        mirror,
+        decal,  // fill outside the domain with transparent
+    };
+
+    /*
+     *  Base class for Render objects that specify the src colors.
+     *
+     *  Shaders are immutable, and sharable between multiple paints, etc.
+     *
+     *  It is common that a shader may be created with a 'localMatrix'. If this is
+     *  not null, then it is applied to the shader's domain before the Renderer's CTM.
+     */
+    class RenderShader : public RefCnt {};
+
+    extern rcp<RenderShader> makeLinearGradient(float sx, float sy,
+                                                float ex, float ey,
+                                                const ColorInt colors[],    // [count]
```

Comment:
[count] meaning we know the size at compile time, like sizeof(colors)/sizeof(ColorInt)? How does this work for non compile time defined arrays?